### PR TITLE
tests/resource/aws_ebs_snapshot_copy: Refactoring to remove inconsistent ProviderFactories usage and make region agnostic

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Noticed this was the only testing left in the provider using the direct `map[string]terraform.ResourceProviderFactory{}` setup instead of `testAccProviderFactories()`. While here, also refactored the testing to make it region agnostic and use consistent test resource naming.

Output from acceptance testing:

```
--- PASS: TestAccAWSEbsSnapshotCopy_disappears (61.46s)
--- PASS: TestAccAWSEbsSnapshotCopy_basic (62.02s)
--- PASS: TestAccAWSEbsSnapshotCopy_withDescription (62.38s)
--- PASS: TestAccAWSEbsSnapshotCopy_withKms (70.19s)
--- PASS: TestAccAWSEbsSnapshotCopy_withRegions (74.09s)
```

### CODE REVIEW

File: aws/resource_aws_ebs_snapshot_copy_test.go [hunk 1]
```diff
@@ -13,7 +12,9 @@ import (
 )
 
 func TestAccAWSEbsSnapshotCopy_basic(t *testing.T) {
-	var v ec2.Snapshot
+	var snapshot ec2.Snapshot
```
Comment (nywilken) [inline aws/resource_aws_ebs_snapshot_copy_test.go]: To me: it would help if variable renaming was done as a separate commit so that the actual change can be validated independently of the renames. I'm now sure how the rest of the folks feel about it.

Comment (bflad) [reply aws/resource_aws_ebs_snapshot_copy_test.go]: Aye - good call. Will try to do that in the future.
